pmkoalas/models/petrinet.py

```diff
@@ -412,13 +420,47 @@ def convert_net_to_xml(net:LabelledPetriNet) -> ET.Element:
             name_node = ET.SubElement(placeNode,'name')
             text_node = ET.SubElement(name_node,'text')
             text_node.text = place.name
+        if (include_prom_bits):
+            if isinstance(place.pid, int):
+                from random import Random
+                rd = Random(place.pid)
+                localNode = str(UUID(int=rd.getrandbits(128), version=4))
+            else:
+                localNode = place.pid
+            prom_node = ET.SubElement(
+                    placeNode, 'toolspecific',
+                    attrib={
+                        'tool' : "ProM",
+                        'version' : "6.4",
+                        'localNodeID' : localNode
+                    }
+                )
+
     for tran in net.transitions:
         tranNode = ET.SubElement(page,'transition', 
                         attrib={'id':"transition-"+str(tran.tid) } )
         if tran.name:
             name_node = ET.SubElement(tranNode,'name')
             text_node = ET.SubElement(name_node,'text')
             text_node.text = tran.name
+        # include additional info for prom
+        if (include_prom_bits):
+            if isinstance(tran.tid, int):
+                from random import Random
+                rd = Random(tran.tid)
+                localNode = str(UUID(int=rd.getrandbits(128), version=4))
```

Comment:
I don't know if the ProM importer will handle non-UUIDs if prom_bits are wanted. 

The initialisation of the pseudo stream could be shifted to a defined seed, and the initialisation for a node could be removed. The seeding of the stream `Random` allows for a deterministic export, which may be helpful for testing and ease of use.

Reply:
How would we confirm whether UUIDs are needed at all? Just load one of the PNs without them in ProM?

Reply:
I don't follow why any of this needs randomness. The nets aren't random: they are exactly as specified. We should let the determinism flow through to get maximum control over the output.

Reply:
I would remove UUID and use the normal id allocated on the transition and places as localNodeID for prom, but prom requires UUID for this element. I will just seed one random stream for the generation though, rather than reseeding using the places and transitions.